tests/cdi/cdi_test.go

```diff
@@ -0,0 +1,135 @@
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool {
+			out, err := RunOcCommand("get", resourceType)
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(out, resourceName)
+		}, time.Duration(2)*time.Minute).Should(BeTrue(), "Timed out waiting for ovm to appear")
+	}
+
+	waitUntilResourceReady := func(resourceName, resourceType, query, expectOut string) {
+		By(fmt.Sprintf("Wait until %s:%s ready", resourceType, resourceName))
+		Eventually(func() bool {
+			//oc get resource with -o jsonpath query
+			out, err := RunOcCommand("get", resourceType, resourceName, query)
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(out, expectOut)
+		}, time.Duration(2)*time.Minute).Should(BeTrue(), "Timed out waiting for vm to appear")
+	}
+
+	Context("will success", func() {
+		It("create pvc should success", func() {
+			overrideTemplateFromParameters(rawPVCFilePath, dstPVCFilePath, sepP, "PVC_NAME="+pvcName, sepP, "EP_URL="+pvcEPHTTPNOAUTHURL)
+			createResourceWithFilePath(pvcName, "persistentvolumeclaim", dstPVCFilePath+".json")
+		})
+
+		It("pvc binding completed", func() {
+			waitUntilResourceReady(pvcName, "pvc", "-o=jsonpath='{.metadata.annotations}'", "pv.kubernetes.io/bind-completed:yes pv.kubernetes.io/bound-by-controller:yes")
+
+		})
+
+		It("create vm should success", func() {
+			overrideTemplateFromParameters(rawVMFilePath, dstVMFilePath, sepP, "VM_NAME="+vmName, sepP, "PVC_NAME="+pvcName)
+			createResourceWithFilePath(vmName, "virtualmachine", dstVMFilePath+".json")
+		})
+
+		It("vm should be running", func() {
+			waitUntilResourceReady(vmName, "vm", "-o=jsonpath='{.status.phase}'", "Running")
+		})
+	})
+})
+
+func WriteJson(name string, json string) (string, error) {
+	jsonFile := fmt.Sprintf("%s.json", name)
+	err := ioutil.WriteFile(jsonFile, []byte(json), 0644)
+	if err != nil {
+		return "", fmt.Errorf("failed to write json file %s", jsonFile)
+	}
+	return jsonFile, nil
+}
+
+func RunOcCommand(args ...string) (string, error) {
+	if KubeVirtOcPath == "" {
+		var err error
+		KubeVirtOcPath, err = exec.LookPath("oc")
+		if err != nil {
+			return "", fmt.Errorf("can not find oc binary")
+		}
+	}
+
+	kubeconfig := flag.Lookup("kubeconfig").Value
+	if kubeconfig == nil || kubeconfig.String() == "" {
+		return "", fmt.Errorf("can not find kubeconfig")
+	}
+
+	master := flag.Lookup("master").Value
+	if master != nil && master.String() != "" {
+		args = append(args, "--server", master.String())
+	}
+	cmd := exec.Command(KubeVirtOcPath, args...)
+	kubeconfEnv := fmt.Sprintf("KUBECONFIG=%s", kubeconfig.String())
+	cmd.Env = append(os.Environ(), kubeconfEnv)
+
+	out, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return "", err
+	}
+	return string(out), nil
+}
+
+func SkipIfNoOc() {
+	if KubeVirtOcPath == "" {
+		var err error
+		KubeVirtOcPath, err = exec.LookPath("oc")
+		if err != nil {
+			Skip("Skip test that requires oc binary")
+		}
+	}
+}
```

Comment:
Let's see if we can share these as utility functions also.

Reply:
yes, you mean move to `kubevirt/kubevirt` right ?

Reply:
No, a utils file in the tests module right here.

Reply:
done